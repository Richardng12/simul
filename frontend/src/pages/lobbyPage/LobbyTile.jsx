import React, { useState } from 'react';
import { useHistory } from 'react-router';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import ClearIcon from '@material-ui/icons/Clear';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import EnterPasswordModal from './enterPasswordModal';
import { deleteLobby } from '../../store/lobby/lobbyActions';
import style from './LobbyTile.module.css';

const LobbyTile = props => {
  const { name, id, isPublic, password, createdBy, userId, deleteLobbyFromDB } = props;
  const history = useHistory();
  const [open, setModal] = useState(false);

  const changeHistory = path => {
    if (isPublic) {
      history.push(path);
    } else {
      setModal(true);
    }
  };

  const handleClose = () => {
    setModal(false);
  };

  return (
    <div>
      <EnterPasswordModal
        open={open}
        onClose={handleClose}
        name={name}
        password={password}
        lobbyId={id}
      />
      {createdBy === userId && (
        <ClearIcon
          onClick={() => {
            deleteLobbyFromDB(id);
          }}
        />
      )}
      {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
      <div className={style.lobbyTile} onClick={() => changeHistory(`/lobby/${id}`)}>
        <div className={style.background}>
          {!isPublic && (
            <LockOutlinedIcon preserveAspectRatio="none" className={style.lockedIcon} />
          )}
        </div>
        <div className={style.bottomSection}>
          <p className={style.lobbyName}>{name}</p>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  userId: state.profileReducer.user._id,
});

const mapDispatchToProps = dispatch => ({
  deleteLobbyFromDB: bindActionCreators(deleteLobby, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(LobbyTile);
