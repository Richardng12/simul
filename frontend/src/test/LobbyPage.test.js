import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import LobbyPage from '../pages/lobbyPage/lobbyPage';
import LobbyTile from '../pages/lobbyPage/LobbyTile';
import AddLobbyModal from '../pages/lobbyPage/addLobbyModal';

const mockStore = configureMockStore();

describe('<LobbyPage />', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      lobbyReducer: {
        lobbies: [],
        loading: false,
        currentLobby: null,
        lobbyId: '',
        users: [],
        currentQueue: [],
      },
      profileReducer: {
        user: {},
        loading: false,
        isAuthenticated: false,
        message: '',
      },
    });
  });

  it('renders lobby page', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <LobbyPage />
        </Provider>
      </MemoryRouter>,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders lobby tile', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <LobbyTile />
        </Provider>
      </MemoryRouter>,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders add lobby modal', () => {
    const component = mount(
      <Provider store={store}>
        <AddLobbyModal open />
      </Provider>,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
