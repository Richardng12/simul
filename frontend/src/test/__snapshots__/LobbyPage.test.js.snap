// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LobbyPage /> renders lobby page 1`] = `
<div
  className="lobbyParent"
>
  <div
    className="headerContainer"
  >
    <div
      className="barContainer"
    >
      <div
        className="bar"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root logoButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              height="100%"
              viewBox="0 0 450 450"
              width="100%"
            >
              <path
                d="M340.681 135.739c-13.152-26.976-35.744-42.112-57.568-56.736-16.288-10.912-31.68-21.216-42.56-35.936l-1.952-2.624c-6.432-8.64-13.696-18.432-14.848-26.656-1.152-8.32-8.704-14.24-16.96-13.76a15.957 15.957 0 00-14.88 15.936v285.12c-13.408-8.128-29.92-13.12-48-13.12-44.096 0-80 28.704-80 64s35.904 64 80 64 80-28.704 80-64V72.955c12.544 13.312 27.136 23.104 41.376 32.64 19.168 12.832 37.28 24.96 46.656 44.192 5.888 12 8.704 25.088 8 36.736-1.76 28.512-12.512 57.184-28.032 74.88-5.856 6.624-5.216 16.736 1.44 22.56 6.592 5.888 16.704 5.184 22.56-1.44 20.032-22.752 33.824-58.784 35.968-94.016 1.055-17.376-2.816-35.616-11.2-52.768z"
              />
            </svg>
          </span>
        </button>
        <p
          className="barTitle"
        >
          LOBBIES
        </p>
        <div
          className="user"
        >
          <p
            className="userName"
          />
        </div>
      </div>
    </div>
  </div>
  <div />
  <div
    className="searchContainer"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text addButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Add Lobby
      </span>
    </button>
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-root-43 searchField"
      height={20}
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
      >
        Search
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input SearchField-input2-42"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-102 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-104"
          >
            <span>
              Search
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    className="lobbyContainer"
  >
    <p>
      No Lobbies :(
    </p>
  </div>
</div>
`;

exports[`<LobbyPage /> renders lobby tile 1`] = `
<div
  className="lobbyContainer"
>
  <div />
  <div />
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root deleteIcon"
    focusable="false"
    onClick={[Function]}
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
  <div
    className="lobbyTile"
    onClick={[Function]}
  >
    <div
      className="background"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root lockedIcon"
        focusable="false"
        preserveAspectRatio="none"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <div
      className="bottomSection"
    >
      <p
        className="lobbyName"
      />
    </div>
  </div>
</div>
`;
